q2_mlab/doctor.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python
+import click
+import os
+from datetime import datetime
+
+
+def sort_result_artifact_filenames(list_of_artifact_filenames):
+    """
+    Sorts the given list of result filenames by parameter index (assumed to be
+    the beginning of the filename, preceding an underscore e.g. 00004_*.qza)
+
+            Parameters:
+                    list_of_artifact_filenames (List): A list of artifact filenames returned by get_results()
+            Returns:
+                    sorted_artifact_filenames (List): Sorted list of the found artifact filenames
+    """
+    temp_list = [(int(f.split("_")[0]), f) for f in list_of_artifact_filenames]
+    temp_list.sort()
+    sorted_artifact_filenames = [f[1] for f in temp_list]
+    return sorted_artifact_filenames
```

Comment:
I think this can be replaced by using `sorted` and the `key` argument:

```python
In [26]: sorted(['0001_xyz.qza', '0003_sawyer.qza', '0002_tom.qza'], key=lambda name: int(name.split('_')[0]))
Out[26]: ['0001_xyz.qza', '0002_tom.qza', '0003_sawyer.qza']
```

Reply:
Thanks, it should be much cleaner now!